static/conformance-view.js

```diff
@@ -0,0 +1,314 @@
+            )
+        );
+
+        newSelector.selectize({
+            sortField: 'name',
+            valueField: 'id',
+            labelField: 'name',
+            searchField: ['name'],
+            options: compilers,
+            items: config.compilerId ? [config.compilerId] : []
+        }).on('change', _.bind(function() {
+            // Hide the results button when a new compiler is selected
+            this.handleStatusIcon(status, {code: 0, text: ""});
+            this.saveState();
+        }, this));
+
+        this.handleStatusIcon(status, config.status);
+        this.handleToolbarUI();
+    };
+
+    Conformance.prototype.removeCompilerSelector = function(cv) {
+        _.each(this.selectorList.children(), function(row) {
+            var child = $(row);
+            if (child.attr("data-cv") == cv) {
+                child.remove();
+            }
+        }, this);
+        this.handleToolbarUI();
+        this.saveState();
+    };
+
+    // TODO: Maybe use premises
+    Conformance.prototype.compile = function(selector) {
+        if (!selector) return;
+        var val = selector.val();
+        if (val) {
+            var cv = Number(selector.attr("data-cv"));
+            var request = {
+                source: this.currentExpandedSource || "",
+                compiler: val,
+                options: this.selectorList.find(".options[data-cv='" + cv + "']").val(),
+                backendOptions: {produceAst: false},
+                filters: {},
+                extras: {
+                    cv: cv,
+                    storeAsm: false,
+                    emitCompilingEvent: false,
+                    ignorePendingRequest: true
+                }
+            };
+            Compiler.Compiler.prototype.sendCompile(request, _.bind(this.onCompileResponse, this));
+        }
+    };
+
+    Conformance.prototype.onEditorChange = function(editorId, newSource) {
+        if (editorId == this.editorId) this.source = newSource;
+    };
+
+    Conformance.prototype.onEditorClose = function(editorId) {
+        if (editorId == this.editorId) {
+            _.defer(function (self) {
+                self.container.close();
+            }, this);
+        }
+    };
+
+    Conformance.prototype.onCompileResponse = function (request, result, cached) {
+        var allText = _.pluck((result.stdout || []).concat(result.stderr || []), 'text').join("\n");
+        var failed = result.code !== 0;
+        var warns = !failed && !!allText;
+        var status = {
+            text: allText,
+            code: failed ? 3 : (warns ? 2 : 1)
+        };
+        this.handleStatusIcon(this.selectorList.find('[data-cv="' + request.extras.cv + '"] .status'), status);
+        this.saveState();
+    };
+
+    Conformance.prototype.compileAll = function() {
+        Compiler.Compiler.prototype.expand(this.source).then(_.bind(function (expanded) {
+            this.currentExpandedSource = expanded;
+            // And trun off every icon
```

Comment:
Nothing now. We used to hide the status of previous compilations upon the start of a new one, but that's now done elsewhere. I'll move this where it is meant to go